drivers/mrf24j40/mrf24j40.c

```diff
@@ -52,13 +51,12 @@ int mrf24j40_reset(mrf24j40_t *dev)
 
     netdev_ieee802154_reset(&dev->netdev);
 
-    /* get an 8-byte unique ID to use as hardware address */
-    luid_get(addr_long.uint8, IEEE802154_LONG_ADDRESS_LEN);
-    addr_long.uint8[0] &= ~(0x01);
-    addr_long.uint8[0] |=  (0x02);
+    /* set device address */
+    netdev_ieee802154_setup(&dev->netdev);
+
     /* set short and long address */
-    mrf24j40_set_addr_long(dev, ntohll(addr_long.uint64.u64));
-    mrf24j40_set_addr_short(dev, ntohs(addr_long.uint16[0].u16));
+    mrf24j40_set_addr_long(dev, (uint64_t*)dev->netdev.long_addr);
```

Comment:
This cast changes the alignment requirements from 1 byte to up to 8 bytes (on 32 bit machines only to 4 bytes). Maybe `mrf24j40_set_addr_long` can just take an `uint8_t` pointer instead?

Reply:
Yes, I also changed it for `mrf24j40_get_addr_long()` while I was at it.